tests/test_async.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+
+import pytest
+
+import wgpu.utils
+from tests.testutils import run_tests
+from wgpu import MapMode, TextureFormat
+from wgpu.backends.wgpu_native import WgpuAwaitable
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("use_async", [False, True])
+async def test_awaitable_async(use_async, loop_scope="function"):
+    count = 0
+
+    def finalizer(i):
+        return i * i
+
+    def callback(i):
+        awaitable.set_result(i)
+
+    def poll_function():
+        nonlocal count
+        count += 1
+        if count >= 3:
+            callback(10)
+
+    awaitable = WgpuAwaitable("test", callback, finalizer, poll_function)
+
+    if use_async:
+        result = await awaitable.wait_async()
```

Comment:
I think we can make this work using `result = await awaitable`. That's how it worked earlier.

Reply:
The problem is that I'm not sure *why* what we had before worked.  

According to [this documentation](https://peps.python.org/pep-0492/#await-expression), the value returned by `__await__` has to be an iterator.  Anything else is a TypeError.  

Likewise, according to [this stackoverflow answer](https://stackoverflow.com/questions/63964011/precise-specification-of-await), the value returned by `__await__` is library specific, and the value returned by trio might be different than the value returned by asyncio.  

But I can't argue with the fact that the code seems to work.  So I'll keep my changes to _api.py, but undo _helper.py to be a mixture of the old and new.  And I'll keep the tests.

New version in a couple of hours.

Reply:
Hmm.  As I feared, the current __await__ is asyncio only.  When I switched test-async to test both asyncio and trio, I got the error message:

```
TypeError: trio.run received unrecognized yield message None. Are you trying to use a library written for some other framework like asyncio? That won't work without some kind of compatibility shim.
```
This failed running with `trio` and calling `async awaitable` where `awaitable` has something similar to the current implementation of __await__ .